lib/external-secret.js

```diff
@@ -1,53 +1,94 @@
 'use strict'
 
+const JSONStream = require('json-stream')
+
 /**
- * Block asynchronous flow.
- * @param {number} milliseconds - Number of milliseconds to block flow operation.
- * @returns {Promise} Promise object representing block flow operation.
+ * Creates an FIFO queue which you can put to and take from.
+ * If theres nothing to take it will wait with resolving until
+ * something is put to the queue.
+ * @returns {Object} Queue instance with put and take methods
  */
-function sleep ({ milliseconds }) {
-  return new Promise(resolve => setTimeout(resolve, milliseconds))
+function createEventQueue () {
+  const queuedEvents = []
+  const waitingResolvers = []
+
+  return {
+    take: () => queuedEvents.length > 0
+      ? Promise.resolve(queuedEvents.shift())
+      : new Promise(resolve => waitingResolvers.push(resolve)),
+    put: (msg) => waitingResolvers.length > 0
+      ? waitingResolvers.shift()(msg)
+      : queuedEvents.push(msg)
+  }
+}
+
+async function startWatcher ({
+  kubeClient,
+  customResourceManifest,
+  logger,
+  eventQueue
+}) {
+  const deathQueue = createEventQueue()
+
+  try {
+    while (true) {
+      logger.info('Starting watch stream')
```

Comment:
To be consistent with stop log
```suggestion
      logger.debug('Starting watch stream')
```